integration/simulation/validate_chain.go

```diff
@@ -377,35 +379,64 @@ func checkLogsReceived(t *testing.T, s *Simulation) {
 		sub.Unsubscribe()
 	}
 
-	for _, channel := range s.LogChannels {
-		checkReceivedHOCAndPOCLogs(t, channel)
+	logsReceived := 0
```

Comment:
In practice, not every subscription was getting a relevant log from both the HOC and POC contracts, due to the randomness in how the sim runs. So now, we check that at least one log is received (i.e. the subscriptions are actually delivering stuff), and then further down we check that for what we do receive, it's relevant and is from the HOC or POC contracts (the only event emitters in the sims).